### PR TITLE
Feature/cvsb 8681

### PR DESCRIPTION
**Pull request by oalexandru**

### CODE REVIEW

**Comment by arupdvsa** [comment]
https://jira.dvsacloud.uk/browse/CVSB-8867. there are obvious serverless-offline limitations additionally. we can address this strategically ... and with a note we are approving this conditionally

File: src/services/RetroGenerationService.ts [hunk 1]
```diff
@@ -71,17 +72,24 @@ class RetroGenerationService {
 
                     }
 
+                    console.log("VIN", testResult.vin);
```
Comment (KHodderDVSA) [inline src/services/RetroGenerationService.ts]: Lots of console logs?

File: src/services/RetroGenerationService.ts [hunk 2]
```diff
@@ -70,18 +70,17 @@ class RetroGenerationService {
...
-
                     if (defectsDetails) {defects = `Defects: ${defectsDetails};\r\n`; }
                     if (testType.reasonForAbandoning) {reasonForAbandoning = `Reason for abandoning: ${testType.reasonForAbandoning};\r\n`; }
                     if (testType.additionalCommentsForAbandon) {additionalCommentsAbandon = `Additional comments for abandon: ${testType.additionalCommentsForAbandon};\r\n`; }
 
                     detailsTemplate.activity.value = (activity.activityType === "visit") ? ActivityType.TEST : ActivityType.WAIT_TIME;
                     detailsTemplate.startTime.value = moment(testResult.testStartTimestamp).tz(TimeZone.LONDON).format("HH:mm:ss");
                     detailsTemplate.finishTime.value = moment(testResult.testEndTimestamp).tz(TimeZone.LONDON).format("HH:mm:ss");
-                    detailsTemplate.vrm.value = testResult.vrm;
+                    detailsTemplate.vrm.value = (testResult.vehicleType === "trl" ) ? testResult.trailerId : testResult.vrm;
```
Comment (arupdvsa) [inline src/services/RetroGenerationService.ts]: pls move trl to enum

File: src/services/RetroGenerationService.ts [hunk 3]
```diff
@@ -70,18 +70,17 @@ class RetroGenerationService {
...
                     detailsTemplate.chassisNumber.value = testResult.vin;
                     detailsTemplate.testType.value = (testType.testCode).toUpperCase();
-                    detailsTemplate.seatsAndAxles.value = (testResult.vehicleType === "psv") ? testResult.numberOfSeats : "" ;
+                    detailsTemplate.seatsAndAxles.value = (testResult.vehicleType === "psv") ? testResult.numberOfSeats : testResult.noOfAxles;
```
Comment (arupdvsa) [inline src/services/RetroGenerationService.ts]: please move psv to enum

**Pull request merged**